PrimeNumbers/PrimeNumber_join.cpp

```diff
@@ -402,22 +412,42 @@ class PrimeNumbers
             JOIN_TYPE = 1;
         }
 
-        if (mwaitx_cycle_count_used)
+        if (wait_count_used)
         {
-            if ((join_type < 3) || (join_type > 6))
+            if ((join_type < 3) || (join_type > 10))
             {
                 printf("Warning: '--mwaitx_cycle_count' is specified, but value will not be used for 'pause' wait type.\n");
             }
             else
             {
-                MWAITX_CYCLES = mwaitx_cycle_count;
+                WAIT_UNITS = wait_count;
             }
         }
         else
         {
-            if ((join_type >= 3) && (join_type <= 6))
+            if ((join_type >= 3) && (join_type <= 10))
             {
-                printf("Warning: '--mwaitx_cycle_count' is needed when join_type is related to mwaitx.\n");
+                printf("Warning: '--mwaitx_cycle_count' is needed when join_type is related to mwaitx/umwait.\n");
+                PrintUsageAndExit();
+            }
+        }
+
+        if (umwait_power_state_used)
+        {
+            if ((join_type < 7) || (join_type > 10))
+            {
+                printf("Warning: '--umwait_power_state' is specified, but value will not be used for non-'umwait' wait type.\n");
+            }
+            else
+            {
+                UMWAIT_LOW_POWER = umwait_power_state == 0;
+            }
+        }
+        else
+        {
+            if ((join_type >= 7) && (join_type <= 10))
+            {
+                printf("Warning: '--umwait_power_state' is needed when join_type is related to umwait.\n");
```

Comment:
same here

Reply:
I forgot to include `--umwait_power_state` in usage. Done.